graphql-parser/src/source.rs

```diff
@@ -1,23 +1,46 @@
+use std::io;
+use std::io::prelude::*;
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
 static DEFAULT_SOURCE_NAME: &'static str = "GraphQL";
 
 #[derive(Debug)]
-pub struct Source<'a> {
-  pub name: &'a str,
-  pub body: &'a str,
+pub struct Source {
+  pub name: String,
+  pub body: String,
 }
 
-impl<'a> Source<'a> {
-  pub fn new(body: &'a str) -> Source<'a> {
+impl Source {
+  pub fn new(body: String) -> Self {
```

Comment:
How about making the signature `pub fn new(name: Option<String>, body: String)`? Or use the setter patter so: `Source::new(body).set_name(name)`. I'd prefer the option approach.